client/wfnews-war/src/main/angular/src/app/components/public-incident-page/incident-maps-panel/incident-maps-panel.component.ts

```diff
@@ -102,14 +103,37 @@ export class IncidentMapsPanel implements OnInit {
   }
 
   downloadMap(mapLink, fileName) {
+    const request = this.generateMapRequest(mapLink)
+    this.fetchMapResponse(request, fileName)
+  }
+
+  generateMapRequest(mapLink) {
     const url = mapLink;
-    const request = this.httpClient.request(
-      new HttpRequest('GET', url, {
-        reportProgress: true,
-        responseType: 'blob',
-      }),
-    );
+    let isMobile = false;
 
+    this.capacitorService.isMobile.then(response => {
```

Comment:
I think "isMobile" will still be false because` this.capacitorService.isMobile.then(response => { isMobile = response; })` is asynchronous.

maybe better to use :

`try {
    isMobile = await this.capacitorService.isMobile();
    } catch (error) {}`

Reply:
Thanks, I added some new changes now